BinaryObjectScanner/FileType/RAR.cs

```diff
@@ -91,5 +103,71 @@ public bool Extract(Stream? stream, string file, string outDir, bool lookForHead
             return false;
 #endif
         }
+#if NET462_OR_GREATER || NETCOREAPP
+        /// <inheritdoc cref="IExtractable.Extract(Stream?, string, string, bool)"/>
+        // Extraction method for non-solid archives. This iterates over each entry in the archive to extract every file
+        // individually, in order to extract all valid files from the archive.
+        public bool Extract_Non_Solid(ReaderOptions? readerOptions, RarArchive? rarFile, string file, string?
+                outDir, bool includeDebug) 
+        {
+
+            
+            foreach (var entry in rarFile.Entries)
+            {
+                try
+                {
+                    // If the entry is a directory
+                    if (entry.IsDirectory)
+                        continue;
+
+                    // If the entry has an invalid key
+                    if (entry.Key == null)
+                        continue;
+
+                    // If we have a partial entry due to an incomplete multi-part archive, skip it
+                    if (!entry.IsComplete)
+                        continue;
+
+                    string tempFile = Path.Combine(outDir, entry.Key);
+                    var directoryName = Path.GetDirectoryName(tempFile);
+                    if (directoryName != null && !Directory.Exists(directoryName))
+                        Directory.CreateDirectory(directoryName);
+
+                    entry.WriteToFile(tempFile);
+                }
+                catch (Exception ex)
+                {
+                    if (includeDebug) Console.WriteLine(ex);
+                }
+            }
+            return true;
+        }
+        
+        /// <inheritdoc cref="IExtractable.Extract(Stream?, string, string, bool)"/>
+        
+        // Extraction method for solid archives. Uses ExtractAllEntries because extraction for solid archives must be
+        // done sequentially, and files beyond a corrupted point in a solid archive will be unreadable anyways.
+        public bool Extract_Solid(RarArchive rarFile, string? outDir, bool includeDebug)
+        {
+                try
+                {
+                    if (outDir != null && !Directory.Exists(outDir))
+                        Directory.CreateDirectory(outDir);
+                    
+                    rarFile.ExtractAllEntries().WriteAllToDirectory(outDir, new ExtractionOptions()
```

Comment:
The extension method `WriteToDirectory` should be able to replace `ExtractAllEntries().WriteAllToDirectory`